### ISSUE TITLE
Clean Up Root Serializer Model

### ISSUE THREAD

**Issue by Mike-EEE** [ExtendedXmlSerializer/ExtendedXmlSerializer#285]
In performing documentation for #283, I encountered a bit of a mindsore at the root of the source project.

I have decided to clean this up, and it will introduce breaking changes _but only if you are an extension developer using one of the dependencies in your extension_.  This does not impact the Configuration API at all and casual users will not (should not 😁) be impacted.

## What Happened?

At some point as I began my first attempt at working on v3, I had moved some code over from it somehow, or shared it.  This was a mistake.  The code was not only poorly designed, but it also was presumptuous.  In essence, I was getting cute with my design and I was fleshing out some components to not only work with XML but also with JSON.

So, you'll see a bunch of oddly-placed code such as the following:

https://github.com/ExtendedXmlSerializer/ExtendedXmlSerializer/blob/f5c67cf131ed039e3e0d7868c637a357c9f7f312/src/ExtendedXmlSerializer/ISerializer.cs#L3-L8

With "implementations" such as this:

https://github.com/ExtendedXmlSerializer/ExtendedXmlSerializer/blob/f5c67cf131ed039e3e0d7868c637a357c9f7f312/src/ExtendedXmlSerializer/Serializer.cs#L5-L22

You'll notice that `System.Xml.XmlWriter` or `System.Xml.XmlReader` are nowhere to be found, and are in fact meant to take the place of the `TWrite` and `TRead` generic arguments from this registration:

https://github.com/ExtendedXmlSerializer/ExtendedXmlSerializer/blob/f5c67cf131ed039e3e0d7868c637a357c9f7f312/src/ExtendedXmlSerializer/ExtensionModel/SerializationExtension.cs#L15-L17

YUCK. 🤮 😆

The code in question was not used much, so again the impact is minimal.  In fact, I only found one test and issue where it was impacted, #246.  Now all this code:

https://github.com/ExtendedXmlSerializer/ExtendedXmlSerializer/blob/ef60368f166d9301186ed7ab723f81b9ea3a01ec/test/ExtendedXmlSerializer.Tests/ReportedIssues/Issue246Tests.cs#L74-L86

Becomes this code:

``` csharp
public IServiceRepository Get(IServiceRepository parameter)
	=> parameter.Decorate<IFormatWriters, Writers>();

void ICommand<IServices>.Execute(IServices parameter) {}

sealed class Writers : IFormatWriters
{
	readonly IFormatWriters _writers;

	public Writers(IFormatWriters writers) => _writers = writers;

	public IFormatWriter Get(XmlWriter parameter) => new Writer(_writers.Get(parameter));
}
```

## Who Does This Impact?

This should only impact extension authors... and I am not sure we have many, hence the need for documentation in case they want to explore. :)

None of this impacts the Configuration API  (the most publicly-facing component of ExtendedXmlSerializer), so chances are that none of this impacts you. 

## What Do I Have To Do?

You shouldn't have to do anything.  Most of the changes impacted internal classes.  In fact, I now see that `IFormatWriters` is an internal class, so the above code for #246 isn't even usable outside of the test project. 😆

However, I am making this post in case someone does run into something and we can have a chat about it.

Finally, part of the rationale is that we're documenting code to make it more consumable by extension authors, so tackling this now is better than later if/when more onboard and can be impacted down the road.

**Comment by Mike-EEE** [comment]
I'm removing the `breaking change` tag as it turns out the dependencies I was concerned about are internal, anyways.

The other change here is that `IExtendedXmlSerializer` has moved from `ExtendedXmlSeralizer.ExtensionModel.Xml.IExtendedXmlSerializer` to `ExtendedXmlSeralizer.IExtendedXmlSerializer`.  This is a namespace change like the extension methods that is not a breaking change.

**Comment by Mike-EEE** [comment]
Resolved via #287
